core/ir/aarch64/codec.txt

```diff
@@ -425,33 +425,38 @@ x0110111xxxxxxxxxxxxxxxxxxxxxxxx  tbnz   tbz
 
 ## Load/store register (unscaled immediate)
 
-00111000000xxxxxxxxx00xxxxxxxxxx  sturb  mem9 : w0  # STURB Wt,[Xn,#simm]
-00111000010xxxxxxxxx00xxxxxxxxxx  ldurb  w0 : mem9
-00111000100xxxxxxxxx00xxxxxxxxxx  ldursb x0 : mem9
-00111000110xxxxxxxxx00xxxxxxxxxx  ldursb w0 : mem9
+# Base
+10111000000xxxxxxxxx00xxxxxxxxxx  stur   mem9 : w0
+11111000000xxxxxxxxx00xxxxxxxxxx  stur   mem9 : x0
+00111000000xxxxxxxxx00xxxxxxxxxx  sturb  mem9 : w0
+01111000000xxxxxxxxx00xxxxxxxxxx  sturh  mem9 : w0
+
+# SIMD
 00111100000xxxxxxxxx00xxxxxxxxxx  stur   mem9 : b0
-00111100010xxxxxxxxx00xxxxxxxxxx  ldur   b0 : mem9
-00111100100xxxxxxxxx00xxxxxxxxxx  stur   mem9 : q0
-00111100110xxxxxxxxx00xxxxxxxxxx  ldur   q0 : mem9
+01111100000xxxxxxxxx00xxxxxxxxxx  stur   mem9 : h0
+10111100000xxxxxxxxx00xxxxxxxxxx  stur   mem9 : s0
+11111100000xxxxxxxxx00xxxxxxxxxx  stur   mem9 : d0
+00111100100xxxxxxxxx00xxxxxxxxxx  stur   mem9q : q0
```

Comment:
It is hard to know what is being fixed here from the diff: it looks like a bunch of lines are just being rearranged.  The commit description mentions 128-bit which I believe would be "q0": does that mean the two "q0" lines are all that changed here?  And the change is s/mem9/mem9q/?  Looking at the bug description: yes the bug is about the memory size.  Maybe add a few words to the commit description ("s/mem9/mem9q/" or sthg) making it easier to know what this is actually changing since it's hard to see from the diff?

Reply:
Yes, the fix is to use the correct mask: mem9q. The re-arrangement was to put the encodings in organised groups and conventional sequences to make maintenance and future changes less error prone. It was quite difficult to read.